pkg/api/types.go

```diff
@@ -1407,6 +1409,69 @@ type ReplicationControllerList struct {
 	Items []ReplicationController `json:"items"`
 }
 
+// DisruptionBudget is an object to define the max disruption that can be caused to a collection of pods
+type DisruptionBudget struct {
+	// Specification of the desired behavior of the DisruptionBudget.
+	Spec DisruptionBudgetSpec `json:"spec,omitempty"`
+	// Most recently observed status of the DisruptionBudget.
+	Status DisruptionBudgetStatus `json:"status,omitempty"`
+}
+
+// DisruptionBudgetSpec is a description of a DisruptionBudget.
+type DisruptionBudgetSpec struct {
+	// RateLimit is a rate limit on disruptions (preemption and other evictions) across the corresponding
+	// collection of pods, e.g. no more than one disruption per hour across the pods of a particular Service
+	RateLimit *RateLimitSpec `json:"ratelimit,omitempty"`
```

Comment:
This type allows only one rate limit.  At Google, we have found it useful to express guarantees over both a 30m and a 1d window.  E.g. the daily budget may allow for 100 disruptions in a day, but in any 30m window we can only cause 7.

I suggest we create the type such that a user can specify a list of rate limits, but also say that we only define what will happen if the user specifies exactly 1.  This lets us add support for multiple rate limits later without changing semantics on any previously supported specs.

@bgrant0607 can you verify this would follow all the API rules?